src/cursor/abstract_cursor.ts

```diff
@@ -413,7 +412,7 @@ export abstract class AbstractCursor<
           // these do need to be transformed since they are copying the rest of the batch
           const internalDocs = (transform
             ? this[kDocuments].splice(0, this[kDocuments].length).map(transform)
-            : this[kDocuments].splice(0, this[kDocuments].length)) as T[]; // TODO(NODE-3283): Improve transform typing
+            : this[kDocuments].splice(0, this[kDocuments].length)) as TSchema[]; // TODO(NODE-3283): Improve transform typing
```

Comment:
That is the type of `this[kDocuments]` although you did make me think of a potential improvement (really just internally for us) but if we instead type `this[kTransform]` to `(doc: TSchema) => TSchema` then we don't need these casts. 

When you pass the transform into `.map()` it will be `(doc: TSchema) => T` but map will return a new `Cursor<T>`. Where `T` is from the result type of the function passed to `.map`. So the argument that goes into `this[kTransform]` is the old schema and it returns the "current" one.

I suppose the most accurate would be `this[kTransform]: (doc: unknown) => TSchema` but then we need equivalent casting/whatever to narrow unknown

Reply:
Well my original comment was because I didn't see a type on kDocument and was thinking about ways to get rid of the `as` because `this[kDocuments].splice` should automatically return something with the type of `<Type of<this[kDocuments]>>[]` so in this case if the source array was already typed, then this would automatically be TSchema[] so the typing here made me curious.

I actually played around a lot with prototyping a change to ktransform and others when thinking about ways to be able to type projections fully, and narrowing that down was like, a several hundred line TS object that existed only for being able to pass around the type accurately, and I think the `unknown` above might run into the same situation?  Although would you be able to use the `Partial` keyword here for some help? I like the idea of the different typing, just wondering if the effort required to type that unknown is worth it

Reply:
Sorry for the lack of reply, been focused on getting this PR ready for fixing the user issues. I think accurately typing the transform is something we should look into, but it would only affect our internal TS code so at the moment we wouldn't need to prioritize it. but going forward more accuracy even if it requires TS helpers etc. could help catch any issues here.